contracts/xasset/src/storage.rs

```diff
@@ -53,4 +53,19 @@ pub struct Interest {
     pub amount: i128,
     /// Amount of interest paid
     pub paid: i128,
-}
\ No newline at end of file
+}
+
+#[contracttype]
+#[derive(Clone, Copy, Default)]
+pub struct InterestDetail {
+    /// Amount of interest accrued
+    pub amount: i128,
+    /// Amount of interest paid
+    pub paid: i128,
+    /// Amount of interest accrued in XLM
+    pub amountInXLM: i128,
+    /// Amount of interest in XLM that will accrue 5 minutes from now
+    pub approvalAmount: i128,
+    /// Unix timestamp of when interest accrual was last calculated
+    pub lastInterestTime: u64,
```

Comment:
Shouldn't these all be snake case?
```suggestion
    pub amount_in_xlm: i128,
    /// Amount of interest in XLM that will accrue 5 minutes from now
    pub approval_amount: i128,
    /// Unix timestamp of when interest accrual was last calculated
    pub last_interest_time: u64,
```